storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -437,7 +457,7 @@ public void close() {
 
                 // Note that internal cache does not require explicit cleaning/closing. We don't want to invalidate or cleanup
                 // the cache as both would lead to triggering of removal listener.
-
+                internalCache.cleanUp();
```

Comment:
this is not required since GC takes care of clean up. In fact, cleanUp is expensive because it performs maintenance activity such as eviction etc which we don't want to do.

Reply:
updated